server/server.go

```diff
@@ -71,7 +71,12 @@ func (s *Server) AddChannel(channel wasabi.Channel) {
 // Run starts the server
 // returns error if server is already running
 // or if server fails to start
-func (s *Server) Run() (err error) {
+// done is optional and can be used to signify that the server is ready for
+// usage by closing it.
+func (s *Server) Run(done ...chan<- struct{}) (err error) {
+	if len(done) > 1 {
+		panic("only one channel can be supplied!")
+	}
```

Comment:
@KianYang-Lee The idea is great, the only one problem I see, not sure we can commit variadic argument to channel type. It'd be nice have possibility to extend this method in future if we'll need to without breaking public interface. If we'll commit to chan now, we'll not have this possibility any more. What do you think of use option functions to accept optional arguments like  `Server.Run(WithNotifyOnStart(done))`? in this case, we'll be able to extend this method in the future.